conversant/prompt_chatbot.py

```diff
@@ -186,20 +187,42 @@ def _update_max_context_examples(
             "tokens"
         )
 
-    def reply(self, query: str) -> Interaction:
-        """Replies to a query given a chat history.
+    def should_stop(self, response: str) -> str:
+        """Given a response, returns the stop sequence it ends with if any.
 
-        The reply is then generated directly from a call to a LLM.
+        Args:
+        response (str): Response coming from prompt chatbot.
+
+        Returns:
+        stop_seq (str): The stop sequence at the end of response.
+
+        """
+        for stop_seq in self.client_config["stop_sequences"]:
+            if response.endswith(stop_seq):
+                return stop_seq
+
+        # If there is not a stop sequence at the end
+        return None
```

Comment:
Should we use an empty string instead of None? Especially since the function signature defines `str` as the return type.

Reply:
Or if you want to keep the None, you can change the return type to `Optional[str]` and import ` from typing import Optional`

Reply:
great suggestions, thanks!